python-checks/src/main/java/org/sonar/python/checks/cdk/PublicNetworkAccessToCloudResourcesCheck.java

```diff
@@ -51,10 +54,58 @@ protected void registerFqnConsumer() {
     checkFqn("aws_cdk.aws_rds.DatabaseInstance", PublicNetworkAccessToCloudResourcesCheck::checkDatabaseInstance);
 
     checkFqn("aws_cdk.aws_rds.CfnDBInstance", (subscriptionContext, callExpression) ->
-      getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresent(
-        argument -> argument.addIssueIf(isTrue(), MESSAGE)
-      )
-    );
+      getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresent(argument -> {
+        if (argument.hasExpression(isTrue()) && !isSafeNetwork(subscriptionContext, callExpression)) {
+          argument.addIssue(MESSAGE);
+        }
+      }));
+  }
+
+  private static boolean isSafeNetwork(SubscriptionContext subscriptionContext, CallExpression callExpression) {
+    Optional<QualifiedExpression> subnetGroupName = getArgument(subscriptionContext, callExpression, "db_subnet_group_name")
+      .flatMap(expressionFlow -> expressionFlow.getExpression(isQualifiedExpression()))
+      .map(QualifiedExpression.class::cast);
+
+    if (subnetGroupName.isEmpty()) {
+      return false;
+    }
+
+    QualifiedExpression qualifiedExpression = subnetGroupName.get();
+    if (!"ref".equals(qualifiedExpression.name().name())) {
+      return false;
+    }
```

Comment:
This only covers the case the SubnetGroup object is referenced. What about
```
rds.CfnDBInstance(
    publicly_accessible=True,
    db_subnet_group_name=rds.CfnDBSubnetGroup(
        subnet_ids=ec2.SubnetType.PUBLIC
    )
)
```
Currently, the implementation is not detecting this simple case.